Makefile

```
@@ -90,13 +90,10 @@ AB-Why3:
 plugins:
 	$(DUNE) build $(DUNE_FLAGS) @$(PLUGINS_DIR)/all
 
-# Alias to build all targets using dune
+# Alias to build all the bytecode artefacts using dune
 # Hopefully more efficient than making "all" depend
-# on "lib" and "bin", since dune can
-# parralelize more
-all:
-	$(DUNE) build $(DUNE_FLAGS) @$(LIB_DIR)/all @$(BTEXT_DIR)/all \
-		@$(PARSERS_DIR)/all @$(BJS_DIR)/all @$(PLUGINS_DIR)/all
+# on "lib" and "bin", since dune can parralelize more
+all: dune build @check
```

This tells `make` that `dune`, `build`, and `@check` are dependencies of the `all` target. Plus, this must use the `$(DUNE)` variable.

This should be:
```suggestion
all:
	$(DUNE) build $(DUNE_FLAGS) @$(LIB_DIR)/check @$(BTEXT_DIR)/check \
		@$(PARSERS_DIR)/check @$(BJS_DIR)/check @$(PLUGINS_DIR)/check
```

---

The following should also work:

```suggestion
all:
	$(DUNE) build $(DUNE_FLAGS) @check
```

---

I agree for `$(DUNE)` and `$(DUNE_FLAGS)` but I don't understand why we should keep enumerate subdirectories? I did that in a old PR only to prevent building tests in the subdirectory `tests`.

If we add another directory at the toplevel, we have to update this list... I doesn't make sense in my opinion.

---

Yes I just replaced `all` in the subcommand. Using naked `@check` should work (I think it will also build the tools but doesn't matter).